src/outputs/restart_hdf5.cpp

```diff
@@ -134,6 +134,16 @@ RestartReaderHDF5::MeshInfo RestartReaderHDF5::GetMeshInfo() const {
   mesh_info.level_gid_lid_cnghost_gflag =
       ReadDataset<int>("/Blocks/loc.level-gid-lid-cnghost-gflag");
 
+  try {
+    mesh_info.derefinement_count = ReadDataset<int>("/Blocks/derefinement_count");
+  } catch (const std::runtime_error &e) {
+    // File does not contain this dataset, so must be older. Set to default value of zero
+    PARTHENON_WARN(
+        "Restarting from an HDF5 file that doesn't contain /Blocks/derefinement_count. \n"
+        "If you are running with AMR, this may cause restarts to not be bitwise exact \n"
+        "with simulations that are run without restarting.");
```

Comment:
Hide this warning behind an `my_rank == 0` check?

Reply:
fixed.